ui/src/cloud/utils/reporting.ts

```diff
@@ -31,6 +31,27 @@ export const updateReportingContext = (properties: KeyValue) => {
   reportingTags = {...reportingTags, ...properties}
 }
 
+// NOTE: typescript can't follow the API results for flags,
+// so we need to convert them to strings here
+const cleanTags = (data: Point): Point => {
+  data.tags = Object.entries(data.tags).reduce((acc, [key, val]) => {
```

Comment:
i was thinking its not too crazy as these are unidirectional data flows where we dont store / reuse the state, and erred on the side of less memory usage, but as we're hardening this part of the code it might be a good idea to reduce the chance of introducing side effect issues. on it.